### PR TITLE
Make driver test capabilities configurable and remove expansion from GKE tests

### PR DESCRIPTION
**Pull request by davidz627**
/kind failing-test

/assign @msau42 @misterikkit

```release-note
NONE
```

### CODE REVIEW

File: test/k8s-integration/driver-config.go [hunk 1]
```diff
@@ -38,10 +40,41 @@ func generateDriverConfigFile(pkgDir, storageClassFile string) (string, error) {
...
 
 	// Fill in template parameters
+	caps := []string{
+		"persistence",
+		"block",
+		"fsGroup",
+		"exec",
+		"multipods",
+		"topology",
+	}
```
Comment (misterikkit) [inline test/k8s-integration/driver-config.go]: Is there a well known source of truth for all valid capability values? Is it part of the CSI capability spec?

File: test/k8s-integration/driver-config.go [hunk 2]
```diff
@@ -38,10 +40,41 @@ func generateDriverConfigFile(pkgDir, storageClassFile string) (string, error) {
...
 
+	/* Unsupported Capabilities:
+	   snapshotDataSource
+	   pvcDataSource
+	   RWX
+	   volumeLimits # PD Supports volume limits but test is very slow
+	   singleNodeVolume
+	   dataSource
+	*/
+
```
Comment (misterikkit) [inline test/k8s-integration/driver-config.go]: :+1:

File: test/k8s-integration/driver-config_test.go [hunk 3]
```diff
@@ -0,0 +1,12 @@
+package main
+
+import "testing"
+
+// WARNING: Not an official test - run this as a quick n dirty way to test your
+// template generation code :)
+func TestGenerateDriverConfigFile(t *testing.T) {
+	_, err := generateDriverConfigFile("/usr/local/google/home/dyzz/go/src/sigs.k8s.io/gcp-compute-persistent-disk-csi-driver", "sc-standard.yaml", "gke")
```
Comment (misterikkit) [inline test/k8s-integration/driver-config_test.go]: Did you want to commit this file as-is?

I am in favor of a test that renders the template and then verifies it parses as valid yaml...

File: test/k8s-integration/driver-config.go (hunk 1 above)
Comment (davidz627) [reply test/k8s-integration/driver-config.go]: They're here - definitely not easy to find.
https://github.com/kubernetes/kubernetes/blob/23320c958edc85feb541db85edc43ec629288a7d/test/e2e/storage/testsuites/testdriver.go#L136

File: test/k8s-integration/driver-config_test.go (hunk 3 above)
Comment (davidz627) [reply test/k8s-integration/driver-config_test.go]: I'm not convinced we need a test for the test. This was just for my personal iteration speed. I would rather remove this file than spend more time on it

File: test/k8s-integration/driver-config.go (hunk 1 above)
Comment (msau42) [reply test/k8s-integration/driver-config.go]: Add a comment where to find the capabilities from?

File: test/k8s-integration/driver-config.go [hunk 4]
```diff
@@ -38,10 +40,41 @@ func generateDriverConfigFile(pkgDir, storageClassFile string) (string, error) {
...
+		"fsGroup",
+		"exec",
+		"multipods",
+		"topology",
+	}
+
+	switch deploymentStrat {
+	case "gke":
+	case "gce":
+		// TODO: OSS K8S supports volume expansion for CSI by default in 1.16+;
```
Comment (msau42) [inline test/k8s-integration/driver-config.go]: Should we consider K8s version specific capability setting? ie if we're running on a 1.16+ cluster, then enable resizing tests.

File: test/k8s-integration/driver-config_test.go (hunk 3 above)
Comment (misterikkit) [reply test/k8s-integration/driver-config_test.go]: :+1:

File: test/k8s-integration/driver-config.go (hunk 4 above)
Comment (misterikkit) [reply test/k8s-integration/driver-config.go]: Do we need it today? Should we try adding it later?

**Comment by davidz627** [comment]
@msau42 @misterikkit updated

File: test/k8s-integration/driver-config.go (hunk 4 above)
Comment (msau42) [reply test/k8s-integration/driver-config.go]: We can do it later but it means that we will be lacking test coverage of these new features we're releasing until GKE can catch up.

Comment (misterikkit) [reply test/k8s-integration/driver-config.go]: Is there a subset of coverage we can get on GKE before k8s 1.16 is available?

**Pull request merged**